services/tenants/transports/apikeys_http.go

```diff
@@ -0,0 +1,207 @@
+			return
+		}
+		if params.Name == "" {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "name cannot be empty",
+			})
+			return
+		}
+
+		if params.TenantID <= 0 {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "organisation_id must be higher than 0",
+			})
+			return
+		}
+
+		if params.ExpirationDate != nil && !params.ExpirationDate.After(time.Now()) {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "expiration_date must be set in the future",
+			})
+			return
+		}
+
+		apiKey, err := t.apiKeySvc.GenerateNewApiKey(params.Name, params.TenantID, params.ExpirationDate)
+		if err != nil {
+			if errors.Is(err, apikeys.ErrTenantIsNotValid) {
+				web.HTTPResponse(w, http.StatusNotFound, web.APIResponseAny{
+					Message: "Organisation does not exist or has been archived",
+				})
+				return
+			} else {
+				web.HTTPError(w, err)
+				return
+			}
+		}
+		web.HTTPResponse(w, http.StatusCreated, Result{
+			ApiKey: apiKey,
+		})
+	}
+}
+
+func (t *APIKeysHTTPTransport) httpValidateApiKey() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		if authHeader == "" || !strings.HasPrefix(authHeader, "Bearer ") {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "Authorization header must be set",
+			})
+			return
+		}
+		idAndKeyCombination := strings.TrimPrefix(authHeader, "Bearer ")
+		validResp, err := t.apiKeySvc.ValidateApiKey(idAndKeyCombination)
+		if err != nil {
+			if errors.Is(err, apikeys.ErrInvalidEncoding) {
+				web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+					Message: "Invalid input",
+				})
+				return
+			} else {
+				web.HTTPError(w, err)
+				return
+			}
+		}
+		if validResp.IsValid {
+			web.HTTPResponse(w, http.StatusOK, validResp)
+			return
+		}
+		web.HTTPResponse(w, http.StatusUnauthorized, web.APIResponseAny{})
+	}
+}
+
+func (t *APIKeysHTTPTransport) httpGetApiKeys() http.HandlerFunc {
+	type Params struct {
+		apikeys.Filter     `pagination:",squash"`
+		pagination.Request `pagination:",squash"`
+	}
+	return func(rw http.ResponseWriter, r *http.Request) {
+		params, err := httpfilter.Parse[Params](r)
+		if err != nil {
+			web.HTTPError(rw, web.NewError(http.StatusBadRequest, "invalid params", ""))
+			return
+		}
+
+		page, err := t.apiKeySvc.ListAPIKeys(params.Filter, params.Request)
+		if err != nil {
+			web.HTTPError(rw, err)
+			return
+		}
+
+		web.HTTPResponse(rw, http.StatusOK, pagination.CreateResponse(r, t.url, *page))
+	}
+}
+
+type apiKeyService interface {
+	ValidateApiKey(base64IdAndKeyCombination string) (apikeys.ApiKeyValidDTO, error)
+	GenerateNewApiKey(name string, tenantId int64, expiry *time.Time) (string, error)
+	RevokeApiKey(base64IdAndKeyCombination string) error
+	ListAPIKeys(filter apikeys.Filter, p pagination.Request) (*pagination.Page[apikeys.ApiKeyDTO], error)
+}
```

Comment:
Transport layer is "higher" level than service so it doesn't need an interface per-se. Could still be useful for tests.

Reply:
discussed, keep as interface for now to make testing easier